sympy/simplify/simplify.py

```diff
@@ -300,6 +300,12 @@ def hypersimp(f, k):
     """
     f = sympify(f)
 
+    if f.has(Piecewise):
+        return None
+    f = f.rewrite(gamma)
+    if f.has(Piecewise):
+        f = f.args[-1][0]
```

Comment:
What if f only has a Piecewise part? Maybe piecewise_fold should be run first.

Reply:
Got your point, but this is not resolving the issue.

Reply:
I did not expect that. Probably dummy_eq has to be improved.

Reply:
This was the expected change right?

```
     if f.has(Piecewise):
+        f = piecewise_fold(f)
         f = f.args[-1][0]

```

Reply:
This is necessary in general because the whole expression f need not be a Piecewise and then the next line will yield a wrong result.

Reply:
Yeah true, so what do you think is causing the dummy_eq to break?

Reply:
> Yeah true, so what do you think is causing the dummy_eq to break?

Why is `dummy_eq` thought to be failing? How is it being used? Can you give the line/test here?

Reply:
> > Yeah true, so what do you think is causing the dummy_eq to break?
> 
> Why is `dummy_eq` thought to be failing? How is it being used? Can you give the line/test here?

Here: https://github.com/sympy/sympy/pull/18696#issuecomment-625774349

Reply:
Actually it is not breaking due to the change in hypersimp(). It was breaking even before this change.